client/components/upload.jsx

```diff
@@ -34,6 +38,27 @@ function FileUpload() {
 
   const handleCancel = () => {
     setFiles({});
+    setShowDropMessage(true);
+  };
+
+  const handleUploadButtonClick = () => {
+    fileInputRef.current.click();
+  };
+
+  const handleFileInputChange = (event) => {
+    const files = event.target.files;
+    for (let i = 0; i < files.length; i++) {
+      addFile(files[i]);
+    }
+  };
+
+  const handleDelete = (url) => {
```

Comment:
Can you clarify the action of handleDelete and cancel? Thanks!

Reply:
Handle delete image by image when trash bin is clicked. Handle cancel deletes all images on upload.